src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -196,6 +202,8 @@ private function checkClass($class, $file = null)
 
             $deprecations = $this->checkAnnotations($refl, $name);
 
+            unset(self::$shortNamespacesCache[$name]);
```

Comment:
a small comment could help understand this line :)
is this covered by a test case?

Reply:
The idea is just to "free" memory because once deprecations have been computed for a class, it has also logically been done for all its ancestors, so we know we are never gonna need it again. Is it relevant / does that make sense to you? I will try to find a good test for this and add it.